src/prfmodel/adapter.py

```diff
@@ -0,0 +1,405 @@
+
+        """
+        parameters = parameters.copy()
+
+        for param in self.parameter_names:
+            parameters[param] = self.inverse_fun(parameters[param])
+
+        return parameters
+
+
+class ParameterConstraint(ParameterTransform):
+    """
+    Constrain parameters to lower or upper bounds.
+
+    Instances of this class can be used inside an `Adapter` object to constrain specific parameters during
+    model fitting using exponential transformation.
+
+    Parameters
+    ----------
+    parameter_names : Sequence of str
+        Names of parameters to be transformed.
+    lower : str or float, optional
+        Lower bound of parameter constraint. If the argument has type `str`, it will use another parameter as the
+        dynamic lower bound. An argument of type `float` will be used as a static lower bound.
+    upper : str or float, optional
+        Upper bound of parameter constraint. If the argument has type `str`, it will use another parameter as the
+        dynamic upper bound. An argument of type `float` will be used as a static upper bound.
+    transform_fun : Callable, optional
+        Function to apply to the lower or upper bound before applying the constraint.
+
+    Examples
+    --------
+    Constrain a parameter to be greater than another parameter.
+
+    >>> import numpy as np
+    >>> import pandas as pd
+    >>> params = pd.DataFrame({
+    >>>     "x": np.array([0.5, 1.0, 1.5]),
+    >>>     "lower_bound": np.array([0.1, 0.2, 0.3])
+    >>> })
+    >>> constraint = ParameterContraint(
+    >>>     parameter_names=["x"],
+    >>>     lower="lower_bound",
+    >>> )
+    >>> params_forward = constraint.forward(params)
+    >>> params_inverse = constraint.inverse(params_forward)
+    >>> assert np.allclose(params_inverse["x"], params["x"])
+
+    Constrain a parameter to be greater than a fixed value.
+
+    >>> constraint = ParameterContraint(
+    >>>     parameter_names=["x"],
+    >>>     lower=1.0,
+    >>> )
+    >>> params_forward = constraint.forward(params)
+    >>> params_inverse = constraint.inverse(params_forward)
+    >>> assert np.allclose(params_inverse["x"], params["x"])
+
+    Constrain a parameter to be greater than the square of another parameter.
+
+    >>> constraint = ParameterContraint(
+    >>>     parameter_names=["x"],
+    >>>     lower="lower_bound",
+    >>>     transform_fun=lambda x: x**2
+    >>> )
+    >>> params_forward = constraint.forward(params)
+    >>> params_inverse = constraint.inverse(params_forward)
+    >>> assert np.allclose(params_inverse["x"], params["x"])
+
+    """
+
+    def __init__(
+        self,
+        parameter_names: Sequence[str],
+        lower: str | float | None = None,
+        upper: str | float | None = None,
+        transform_fun: Callable | None = None,
+    ):
+        forward_fun = ops.exp
+        inverse_fun = ops.log
+
+        super().__init__(parameter_names, forward_fun, inverse_fun)
+
```

Comment:
should there a check that, if the upper/lower is `str`, that it can be found in the parameters? I just tweaked the test for `test_parameter_constraint_upper_transform` with a `upper = "a"` and got a bunch of `KeyError`s.

Reply:
I added a function to check this!